src/Gaffer/MessageHandler.cpp

```diff
@@ -0,0 +1,104 @@
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Image Engine Design Inc nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/MessageHandler.h"
+
+#include "Gaffer/Context.h"
+#include "Gaffer/Plug.h"
+#include "Gaffer/Process.h"
+
+#include "IECore/FilteredMessageHandler.h"
+#include "IECore/VectorTypedData.h"
+
+#include "boost/algorithm/string/join.hpp"
+#include "boost/range/adaptor/transformed.hpp"
+
+#include <sstream>
+
+using namespace stdl
+
+using boost::adaptors::transformed;
+using boost::algorithm::join;
+
+namespace
+{
+	IECore::InternedString g_scenePath("scene:path");
+	IECore::InternedString g_frame( "frame" );
+};
+
+namespace Gaffer
+{
+
+class MessageHandler : public IECore::FilteredMessageHandler
+{
+	public:
+
+		IE_CORE_DECLAREMEMBERPTR( MessageHandler );
+
+		MessageHandler( IECore::MessageHandlerPtr handler ) : IECore::FilteredMessageHandler( handler )
+		{
+		}
+
+		~MessageHandler() override
+		{
+		}
+
+		void handle( Level level, const string &context, const string &message ) override
+		{
+			if( const Process *p = Process::current() )
+			{
+				stringstream ss;
+				ss << context;
+
+				ss << " [ plug: '" << p->plug()->fullName() << "' path: '";
+				auto path = p->context()->get<vector<IECore::InternedString> >( g_scenePath, vector<IECore::InternedString>() );
+				ss << boost::algorithm::join( path | transformed([](const IECore::InternedString& s) -> const string& { return ( const string& ) s;}), ".");
```

Comment:
I'm expecting paths of the form `/group/sphere` but this looks like it's going to make `path.sphere` (no root, wrong separator) - is that correct? I have to confess this new piped ranges thing is new to me...

Reply:
corrected in a21c00b